src/repositories/JobRepository.ts

```diff
@@ -1,146 +1,267 @@
-    },
-    {
-      id: "job_004",
-      jobRoleName: "Junior Data Analyst",
-      description:
-        "Entry-level position for someone passionate about data analysis and insights. Great opportunity to start your data career.",
-      responsibilities: [
-        "Create reports and dashboards",
-        "Analyze business data trends",
-        "Support senior analysts with projects",
-        "Learn data visualization tools",
-      ],
-      jobSpecLink:
-        "https://company.sharepoint.com/sites/hr/jobspecs/junior-analyst.pdf",
-      location: "Birmingham, UK",
-      capability: Capability.DATA,
-      band: Band.E1,
-      closingDate: new Date("2024-10-25"),
-      status: JobStatus.DRAFT,
-      numberOfOpenPositions: 3,
-    },
-  ];
-
-  // Get all jobs
-  getAllJobs(): Job[] {
-    return [...this.jobs]; // Return a copy to prevent external modification
-  }
+import { type Band, type Capability, JobStatus } from "../models/JobModel.js";
 
-  // Get job by ID
-  getJobById(id: string): Job | null {
-    return this.jobs.find((job) => job.id === id) || null;
-  }
-}
+// Infer the database row type from the schema
+type JobRow = InferSelectModel<typeof jobsTable>;
 
-// Singleton instance for the temporary data store
-const jobStore = new InMemoryJobStore();
+// Mapper function to convert database rows to Job model
+function mapJobRowToJob(row: JobRow): Job {
+  return {
+    id: row.id.toString(),
+    jobRoleName: row.jobRoleName,
+    description: row.description,
+    responsibilities: row.responsibilities
+      .split(",")
+      .map((item) => item.trim()),
+    jobSpecLink: row.jobSpecLink,
+    location: row.location,
+    capability: row.capability as Capability,
+    band: row.band as Band,
+    closingDate: new Date(row.closingDate),
+    status: row.status as JobStatus,
+    numberOfOpenPositions: row.numberOfOpenPositions,
+  };
+}
 
-export class JobRepository {
-  // TEMPORARY: These methods use in-memory storage
-  // TODO: Replace with actual database calls when database is ready
+class DatabaseJobStore {
+  private client = createClient({
+    url: "file:jobApp.db",
+  });
+  private db = drizzle(this.client);
 
   async getAllJobs(): Promise<Job[]> {
-    return jobStore.getAllJobs();
+    const rows = await this.db.select().from(jobsTable).all();
+    return rows.map(mapJobRowToJob);
   }
 
   async getJobById(id: string): Promise<Job | null> {
-    return jobStore.getJobById(id);
+    const row = await this.db
+      .select()
+      .from(jobsTable)
+      .where(eq(jobsTable.id, Number.parseInt(id, 10)))
+      .get();
+    return row ? mapJobRowToJob(row) : null;
   }
 
   async getFilteredJobs(filters: JobFilters): Promise<PaginatedJobResponse> {
-    // SQL filtering query will go here when database is implemented
-    // For now, return empty response structure
+    const {
+      capability,
+      band,
+      location,
+      status,
+      search,
+      closingDateFrom,
+      closingDateTo,
```

Comment:
I'm not sure we need to filter by this, so long as it's open, we'd want to see it on a job application site